### PR TITLE
`Exam mode`: Fix breadcrumbs in test exams

### PR DESCRIPTION
**Pull request by manuelmanso**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Fixes issue no. 3, no. 21 and no.24 found during [exam mode testing](https://confluence.ase.in.tum.de/pages/viewpage.action?pageId=138610767).

### Description

Changes the navbar breadcrumbs to translate and show 'TestExam' with the right url for issue 3 and 21.
Bugfix to correctly set the testExam property in the exam-participation component instead of relying on the url params for issue 24.
Change the url and corresponding breadcrumb when starting a test exam from 'new' to 'start'

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students

1. Log in to Artemis
2. Create Test Exam
3. Participate in the test exam and make sure the issues described don't happen anymore
4. Make sure participations in test runs and in non-test exams are unaffected

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-participation.component.ts | 65% | 76% |

### Screenshots

Changed url and breadcrumbs

#### Before

image

#### After

startBreadCrumb

### CODE REVIEW

File: src/main/webapp/app/exam/participate/exam-participation.component.ts [hunk 1]
```diff
@@ -304,7 +304,7 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
             const exerciseIds = exercises.map((exercise) => exercise.id).filter(Number) as number[];
             this.examParticipationService.setExamExerciseIds(exerciseIds);
             // set endDate with workingTime
-            if (!!this.testRunId || this.testExam) {
+            if (!!this.testRunId || this.studentExam?.exam?.testExam) {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exam/participate/exam-participation.component.ts:307]: Is it possible that `this.testExam` also has the wrong value in other places of this component?

Comment (manuelmanso) [reply src/main/webapp/app/exam/participate/exam-participation.component.ts:307]: https://github.com/ls1intum/Artemis/blob/d5cd1f7c87aa1128443cc396cf8a6d8adc14df62/src/main/webapp/app/exam/participate/exam-participation.component.html#L135-L138
Only here in the template, but there is no exam there so this doesn't apply

**Review by ge65cer — CHANGES_REQUESTED** [review]
Tested in testing session on TS1, bug is fixed.
The breadcrumbs should say, e.g., "start" instead of "new".
As discussed during the call, maybe you can also change the URL to list the participation id instead of still having "/new" there, once the exam is started.

Screenshot 2022-10-13 at 14 17 03

**Review by julian-christl — APPROVED** [review]
Tested on TS1, fix work as intended

**Review by stefanwaldhauser — APPROVED** [review]
Looks good enough for me to fix the bug

File: src/test/javascript/spec/component/exam/participate/exam-participation.component.spec.ts [hunk 2]
```diff
@@ -351,6 +351,12 @@ describe('ExamParticipationComponent', () => {
         expect(secondSubmission.isSynced).toBeTrue();
         expect(secondSubmission.submitted).toBeFalse();
 
+        if (!!comp.testRunId || comp.testExam) {
```
Comment (JohannesStoehr) [inline src/test/javascript/spec/component/exam/participate/exam-participation.component.spec.ts:354]: Should you rely on the values in the tested component to determine the correct assertion?
Could you use the parameter of this method instead?

**Review by JohannesStoehr — APPROVED** [review]
Code changes look good. Left one comment about the test case:

File: src/test/javascript/spec/component/exam/participate/exam-participation.component.spec.ts (hunk 2 above)
Comment (manuelmanso) [reply src/test/javascript/spec/component/exam/participate/exam-participation.component.spec.ts]: Fixed, thanks!

**Review by JohannesStoehr — APPROVED** [review]
Thanks for fixing the tests, code and test look good to me now

**Review by ge65cer — APPROVED** [review]
Thanks for applying my suggestions. Approved!

**Pull request merged**